solr/core/src/java/org/apache/solr/storage/SizeAwareDirectory.java

```diff
@@ -318,13 +381,13 @@ public long getChecksum() throws IOException {
     @Override
     public void writeByte(byte b) throws IOException {
       backing.writeByte(b);
-      sizeWriter.apply(1, name);
+      sizeWriter.apply(1, 1, name);
     }
 
     @Override
     public void writeBytes(byte[] b, int offset, int length) throws IOException {
       backing.writeBytes(b, offset, length);
-      sizeWriter.apply(length, name);
+      sizeWriter.apply(length, length, name);
```

Comment:
Ah and now I realize these are wrong. To account for compressing `dump()`, the filePointer before `writeBytes` subtracted from the filePointer after `writeBytes` should work right? If there is a compression, then that would be negative, which should still be accurate?